src/types/dtypes.ts

```diff
@@ -235,7 +235,9 @@ export class DType {
   public static byteArrToString(arr: NumberArray): string {
     let result = "";
     for (let i = 0; i < arr.length; i++) {
-      result += String.fromCharCode(Number(arr[i]));
+      if (Number(arr[i]) !== 0) {
+        result += String.fromCharCode(Number(arr[i]));
```

Comment:
Do you think it makes sense to also else..break here to treat the first `\0` as a terminator? This would save iterating through the whole array when not needed.

Reply:
@Araneidae advises this is a good idea.

Reply:
Yes, I think it would make a lot of sense to simply treat `'\0'` as a terminator.  Then you can use `strnlen()` to measure the length of the string before processing it.

Reply:
Yes good idea. I will make that change now and push the update.